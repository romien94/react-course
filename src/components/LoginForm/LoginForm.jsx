import React from "react";
import { connect } from "react-redux";
import { authenticate } from "../../modules/actions";
import PropTypes from "prop-types";
import {Link} from 'react-router-dom';


import FormLabel from "../common/FormLabel";
import Input from "../common/Input";
import Button from "../common/Button";

class LoginForm extends React.Component {
  state = {
    email: "",
    password: "",
  };

  static propTypes = {
    authenticate: PropTypes.func,
  };
  render() {
    return (
      <form
        onSubmit={async (e) => {
          e.preventDefault();
          const { email, password } = e.target;
          this.props.authenticate(email.value, password.value);
        }}
        className="app-form"
      >
        <div className="app-form__wrapper">
          <h2 className="app-form__title">Войти</h2>
          <p className="app-form__suggestion">
            Новый пользователь?
            <Link to="/register">Зарегистрироваться</Link>
            {/* <a href="" className="app-form__link">
              Зарегистрироваться
            </a> */}
          </p>
          <div className="app-form__fields">
            <div className="app-form__row">
              <FormLabel>
                <span className="app-form__fieldname">Имя пользователя</span>
                <Input
                  type="text"
                  className="app-form__input"
                  name="email"
                  data-testid="username"
                />
              </FormLabel>
            </div>
            <div className="app-form__row">
              <FormLabel>
                <span className="app-form__fieldname">Пароль</span>
                <Input
                  type="password"
                  className="app-form__input"
                  name="password"
                  data-testid="password"
                />
              </FormLabel>
            </div>
            <div className="app-form__row">
              <Button
                type="submit"
                className="button app-form__button"
                data-testid="button"
              >
                Войти
              </Button>
            </div>
          </div>
        </div>
      </form>
    );
  }
}

export default connect(null, { authenticate })(LoginForm);
