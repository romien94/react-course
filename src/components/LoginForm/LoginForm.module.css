.appForm {
    position: relative;
    display: flex;
    flex-direction: column;
    max-width: 505px;
    min-height: 475px;
    background-color: white;
    padding: 65px 60px;
    color: #747474;
    font-size: 15px;
}

.formWrapper {
    width: 100%;
    height: 100%;
    flex: 1;
}

.appForm:before {
    position: absolute;
    top: 50%;
    right: 100%;
    content: '';
    display: block;
    background-image: url('../../assets/images/logo.png');
    background-repeat: no-repeat;
    width: 160px;
    height: 50px;
    transform: translate(-50%, -50%);
}

.formTitle {
    font-size: 36px;
    margin-bottom: 20px;
    color: black;
}

.formSuggestion {
    font-size: 16px;
    font-weight: normal;
    margin-bottom: 20px;
}

.formFields {
    display: flex;
    flex-direction: column;
}

.formRow {
    display: flex;
    margin-bottom: 35px;
}

.formRow:last-child {
    margin-bottom: 0;
    margin-top: auto;
}

.formLink {
    color: #0074ea;
    text-decoration: none;
}

.formLabel {
    display: flex;
    flex-direction: column;
    flex: 1;
}

.formInput {
    border: none;
    background-color: transparent;
    border-bottom: 1px solid gray;
    padding: 7px 0;
    font-size: 15px;
}

.formLabel:hover .formInput {
    border-bottom-color: #0074ea;    
}

.formInput:active,
.formInput:focus {
    outline: none;
    border-bottom-color: #ffc700;
}


