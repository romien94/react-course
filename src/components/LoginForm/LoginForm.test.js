import React from "react";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import LoginForm from "./LoginForm";

describe("Login works fine", () => {
  const { getByTestId } = render(<LoginForm />);
  it("Snapshot matches", () => {
    const mockLoginForm = renderer.create(<LoginForm />).toJSON();
    expect(mockLoginForm).toMatchSnapshot();
  });

  describe("Contains all the necessary inputs with basic setup", () => {
    expect(getByTestId("username")).toBeTruthy();
    expect(getByTestId("password")).toBeTruthy();
    expect(getByTestId("button")).toBeTruthy();
    expect(getByTestId("button")).toHaveAttribute('type','submit');
  });
});