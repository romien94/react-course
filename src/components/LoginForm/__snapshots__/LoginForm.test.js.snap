// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login works fine Snapshot matches 1`] = `
<form
  className="app-form"
  onSubmit={[Function]}
>
  <div
    className="app-form__wrapper"
  >
    <h2
      className="app-form__title"
    >
      Войти
    </h2>
    <p
      className="app-form__suggestion"
    >
      Новый пользователь?
      <a
        className="app-form__link"
        href=""
      >
        Зарегистрироваться
      </a>
    </p>
    <div
      className="app-form__fields"
    >
      <div
        className="app-form__row"
      >
        <label
          className="MuiFormLabel-root"
        >
          <span
            className="app-form__fieldname"
          >
            Имя пользователя
          </span>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline app-form__input"
            data-testid="username"
            onClick={[Function]}
          >
            <input
              className="MuiInputBase-input MuiInput-input"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              type="text"
            />
          </div>
        </label>
      </div>
      <div
        className="app-form__row"
      >
        <label
          className="MuiFormLabel-root"
        >
          <span
            className="app-form__fieldname"
          >
            Пароль
          </span>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline app-form__input"
            data-testid="password"
            onClick={[Function]}
          >
            <input
              className="MuiInputBase-input MuiInput-input"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              type="password"
            />
          </div>
        </label>
      </div>
      <div
        className="app-form__row"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained button app-form__button"
          data-testid="button"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          <span
            className="MuiButton-label"
          >
            Войти
          </span>
        </button>
      </div>
    </div>
  </div>
</form>
`;
